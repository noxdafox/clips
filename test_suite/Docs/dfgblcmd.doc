;;;************************************************************
;;; DEFGLOBAL COMMAND TESTING
;;;
;;; The file dfgblcmd.tst tests the commands described in the
;;; BPG in section 11.8 (defglobal commands).
;;;
;;; To test, execute the command (batch "dfgblcmd.tst").
;;; When the test is complete, check the file dfgblcmd.rsl
;;; for the results (differences between expected and 
;;; actual output are listed).
;;;************************************************************
