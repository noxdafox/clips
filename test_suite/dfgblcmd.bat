(clear)
(defglobal ?*d1* = 1)
(defglobal ?*d2* = 2)
(defglobal ?*d3* = 3)
(defglobal ?*d4* = 4)
(defglobal ?*d5* = 5)
(defglobal ?*d6* = 6)
(defglobal ?*** = 7)
(list-defglobals d6)
(list-defglobals)
(ppdefglobal d1)
(ppdefglobal d2)
(ppdefglobal d3)
(ppdefglobal d4)
(ppdefglobal d5)
(ppdefglobal d6)
(undefglobal)
(undefglobal 10)
(undefglobal bad-name)
(undefglobal d6 10)
(list-defglobals)
(undefglobal d1)
(undefglobal d2)
(undefglobal d3)
(list-defglobals)
(undefglobal *)
(list-defglobals)
(undefglobal *)
(list-defglobals)
(clear)
(defglobal ?*x* = 3 ?*y* = ?*x* ?*z* = (+ 4 ?*x*))
(defglobal ?*q* = (create$ a b c))
(reset)
(show-defglobals x y)
(show-defglobals)
(ppdefglobal x)
(ppdefglobal y)
(ppdefglobal z)
(ppdefglobal q)
(get-reset-globals)
(get-reset-globals TRUE)
(set-reset-globals)
(set-reset-globals TRUE FALSE)
(set-reset-globals 10)
(get-reset-globals)
(set-reset-globals FALSE)
(get-reset-globals)
(bind ?*x* a)
(bind ?*y* 3.2)
(bind ?*z* [x])
(bind ?*q* "red")
(reset)
(show-defglobals)
(set-reset-globals TRUE)
(get-reset-globals)
(reset)
(show-defglobals)
(clear)
(show-defglobals)
(ppdefglobal x)
(clear) ; Some new 6.0 features
(defglobal BOGUS d5 = 5)
(defmodule FOO)
(defglobal FOO ?*d1* = 1)
(defglobal FOO ?*d2* = 2)
(defmodule BAR)
(defglobal BAR ?*d2* = 2)
(defglobal BAR ?*d3* = 3)
(list-defglobals BOGUS)
(list-defglobals FOO 7)
(list-defglobals FOO)
(list-defglobals BAR)
(list-defglobals)
(list-defglobals *)
(get-defglobal-list 7)
(get-defglobal-list BOGUS 7)
(get-defglobal-list)
(get-defglobal-list BOGUS)
(get-defglobal-list FOO)
(get-defglobal-list BAR)
(get-defglobal-list *)
(ppdefglobal d1)
(ppdefglobal d2)
(ppdefglobal d3)
(ppdefglobal BOGUS::d2)
(ppdefglobal FOO::d1)
(ppdefglobal FOO::d2)
(ppdefglobal FOO::d3)
(ppdefglobal BAR::d1)
(ppdefglobal BAR::d2)
(ppdefglobal BAR::d3)
(defglobal-module)
(defglobal-module bogus)
(defglobal-module d3 7)
(defglobal-module d2)
(defglobal-module d3)
(defglobal-module FOO::d1)
(defglobal-module FOO::d2)
(defglobal-module BAR::d2)
(defglobal-module BAR::d3)
(show-defglobals)
(show-defglobals 7)
(show-defglobals BOGUS)
(show-defglobals FOO 7)
(show-defglobals FOO)
(show-defglobals BAR)
(show-defglobals *)
(undefglobal BOGUS::d2)
(undefglobal FOO::d2)
(list-defglobals *)
(undefglobal BAR::d2)
(list-defglobals *)
(undefglobal *)
(list-defglobals *)
