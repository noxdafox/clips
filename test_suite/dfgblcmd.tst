(unwatch all)
(clear)
(dribble-on "Actual//dfgblcmd.out")
(batch "dfgblcmd.bat")
(dribble-off)
(clear)
(open "Results//dfgblcmd.rsl" dfgblcmd "w")
(load "compline.clp")
(printout dfgblcmd "dfgblcmd.bat differences are as follows:" crlf)
(compare-files "Expected//dfgblcmd.out" "Actual//dfgblcmd.out" dfgblcmd)
(close dfgblcmd)
