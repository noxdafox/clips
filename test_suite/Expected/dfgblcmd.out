TRUE
CLIPS> (batch "dfgblcmd.bat")
TRUE
CLIPS> (clear)
CLIPS> (defglobal ?*d1* = 1)
CLIPS> (defglobal ?*d2* = 2)
CLIPS> (defglobal ?*d3* = 3)
CLIPS> (defglobal ?*d4* = 4)
CLIPS> (defglobal ?*d5* = 5)
CLIPS> (defglobal ?*d6* = 6)
CLIPS> (defglobal ?*** = 7)
CLIPS> (list-defglobals d6)
[ARGACCES5] Function list-defglobals expected argument #1 to be of type defmodule name
CLIPS> (list-defglobals)
d1
d2
d3
d4
d5
d6
*
For a total of 7 defglobals.
CLIPS> (ppdefglobal d1)
(defglobal MAIN ?*d1* = 1)
CLIPS> (ppdefglobal d2)
(defglobal MAIN ?*d2* = 2)
CLIPS> (ppdefglobal d3)
(defglobal MAIN ?*d3* = 3)
CLIPS> (ppdefglobal d4)
(defglobal MAIN ?*d4* = 4)
CLIPS> (ppdefglobal d5)
(defglobal MAIN ?*d5* = 5)
CLIPS> (ppdefglobal d6)
(defglobal MAIN ?*d6* = 6)
CLIPS> (undefglobal)
[ARGACCES4] Function undefglobal expected exactly 1 argument(s)
CLIPS> (undefglobal 10)
[ARGACCES5] Function undefglobal expected argument #1 to be of type symbol
CLIPS> (undefglobal bad-name)
[PRNTUTIL1] Unable to find defglobal bad-name.
CLIPS> (undefglobal d6 10)
[ARGACCES4] Function undefglobal expected exactly 1 argument(s)
CLIPS> (list-defglobals)
d1
d2
d3
d4
d5
d6
*
For a total of 7 defglobals.
CLIPS> (undefglobal d1)
CLIPS> (undefglobal d2)
CLIPS> (undefglobal d3)
CLIPS> (list-defglobals)
d4
d5
d6
*
For a total of 4 defglobals.
CLIPS> (undefglobal *)
CLIPS> (list-defglobals)
d4
d5
d6
For a total of 3 defglobals.
CLIPS> (undefglobal *)
CLIPS> (list-defglobals)
CLIPS> (clear)
CLIPS> (defglobal ?*x* = 3 ?*y* = ?*x* ?*z* = (+ 4 ?*x*))
CLIPS> (defglobal ?*q* = (create$ a b c))
CLIPS> (reset)
CLIPS> (show-defglobals x y)
[ARGACCES4] Function show-defglobals expected no more than 1 argument(s)
CLIPS> (show-defglobals)
?*x* = 3
?*y* = 3
?*z* = 7
?*q* = (a b c)
CLIPS> (ppdefglobal x)
(defglobal MAIN ?*x* = 3)
CLIPS> (ppdefglobal y)
(defglobal MAIN ?*y* = ?*x*)
CLIPS> (ppdefglobal z)
(defglobal MAIN ?*z* = (+ 4 ?*x*))
CLIPS> (ppdefglobal q)
(defglobal MAIN ?*q* = (create$ a b c))
CLIPS> (get-reset-globals)
TRUE
CLIPS> (get-reset-globals TRUE)
[ARGACCES4] Function get-reset-globals expected exactly 0 argument(s)
CLIPS> (set-reset-globals)
[ARGACCES4] Function set-reset-globals expected exactly 1 argument(s)
CLIPS> (set-reset-globals TRUE FALSE)
[ARGACCES4] Function set-reset-globals expected exactly 1 argument(s)
CLIPS> (set-reset-globals 10)
TRUE
CLIPS> (get-reset-globals)
TRUE
CLIPS> (set-reset-globals FALSE)
TRUE
CLIPS> (get-reset-globals)
FALSE
CLIPS> (bind ?*x* a)
a
CLIPS> (bind ?*y* 3.2)
3.2
CLIPS> (bind ?*z* [x])
[x]
CLIPS> (bind ?*q* "red")
"red"
CLIPS> (reset)
CLIPS> (show-defglobals)
?*x* = a
?*y* = 3.2
?*z* = [x]
?*q* = "red"
CLIPS> (set-reset-globals TRUE)
FALSE
CLIPS> (get-reset-globals)
TRUE
CLIPS> (reset)
CLIPS> (show-defglobals)
?*x* = 3
?*y* = 3
?*z* = 7
?*q* = (a b c)
CLIPS> (clear)
CLIPS> (show-defglobals)
CLIPS> (ppdefglobal x)
[PRNTUTIL1] Unable to find defglobal x.
CLIPS> (clear) ; Some new 6.0 features
CLIPS> (defglobal BOGUS d5 = 5)
[PRNTUTIL1] Unable to find defmodule BOGUS.

ERROR:
(defglobal BOGUS
CLIPS> (defmodule FOO)
CLIPS> (defglobal FOO ?*d1* = 1)
CLIPS> (defglobal FOO ?*d2* = 2)
CLIPS> (defmodule BAR)
CLIPS> (defglobal BAR ?*d2* = 2)
CLIPS> (defglobal BAR ?*d3* = 3)
CLIPS> (list-defglobals BOGUS)
[ARGACCES5] Function list-defglobals expected argument #1 to be of type defmodule name
CLIPS> (list-defglobals FOO 7)
[ARGACCES4] Function list-defglobals expected no more than 1 argument(s)
CLIPS> (list-defglobals FOO)
d1
d2
For a total of 2 defglobals.
CLIPS> (list-defglobals BAR)
d2
d3
For a total of 2 defglobals.
CLIPS> (list-defglobals)
d2
d3
For a total of 2 defglobals.
CLIPS> (list-defglobals *)
MAIN:
FOO:
   d1
   d2
BAR:
   d2
   d3
For a total of 4 defglobals.
CLIPS> (get-defglobal-list 7)
[ARGACCES5] Function get-defglobal-list expected argument #1 to be of type symbol
CLIPS> (get-defglobal-list BOGUS 7)
[ARGACCES4] Function get-defglobal-list expected no more than 1 argument(s)
CLIPS> (get-defglobal-list)
(d2 d3)
CLIPS> (get-defglobal-list BOGUS)
[ARGACCES5] Function get-defglobal-list expected argument #1 to be of type defmodule name
()
CLIPS> (get-defglobal-list FOO)
(d1 d2)
CLIPS> (get-defglobal-list BAR)
(d2 d3)
CLIPS> (get-defglobal-list *)
(FOO::d1 FOO::d2 BAR::d2 BAR::d3)
CLIPS> (ppdefglobal d1)
[PRNTUTIL1] Unable to find defglobal d1.
CLIPS> (ppdefglobal d2)
(defglobal BAR ?*d2* = 2)
CLIPS> (ppdefglobal d3)
(defglobal BAR ?*d3* = 3)
CLIPS> (ppdefglobal BOGUS::d2)
[PRNTUTIL1] Unable to find defglobal BOGUS::d2.
CLIPS> (ppdefglobal FOO::d1)
(defglobal FOO ?*d1* = 1)
CLIPS> (ppdefglobal FOO::d2)
(defglobal FOO ?*d2* = 2)
CLIPS> (ppdefglobal FOO::d3)
[PRNTUTIL1] Unable to find defglobal FOO::d3.
CLIPS> (ppdefglobal BAR::d1)
[PRNTUTIL1] Unable to find defglobal BAR::d1.
CLIPS> (ppdefglobal BAR::d2)
(defglobal BAR ?*d2* = 2)
CLIPS> (ppdefglobal BAR::d3)
(defglobal BAR ?*d3* = 3)
CLIPS> (defglobal-module)
[ARGACCES4] Function defglobal-module expected exactly 1 argument(s)
CLIPS> (defglobal-module bogus)
[PRNTUTIL1] Unable to find defglobal bogus.
FALSE
CLIPS> (defglobal-module d3 7)
[ARGACCES4] Function defglobal-module expected exactly 1 argument(s)
CLIPS> (defglobal-module d2)
BAR
CLIPS> (defglobal-module d3)
BAR
CLIPS> (defglobal-module FOO::d1)
FOO
CLIPS> (defglobal-module FOO::d2)
FOO
CLIPS> (defglobal-module BAR::d2)
BAR
CLIPS> (defglobal-module BAR::d3)
BAR
CLIPS> (show-defglobals)
?*d2* = 2
?*d3* = 3
CLIPS> (show-defglobals 7)
[ARGACCES5] Function show-defglobals expected argument #1 to be of type symbol
CLIPS> (show-defglobals BOGUS)
[ARGACCES5] Function show-defglobals expected argument #1 to be of type defmodule name
CLIPS> (show-defglobals FOO 7)
[ARGACCES4] Function show-defglobals expected no more than 1 argument(s)
CLIPS> (show-defglobals FOO)
?*d1* = 1
?*d2* = 2
CLIPS> (show-defglobals BAR)
?*d2* = 2
?*d3* = 3
CLIPS> (show-defglobals *)
MAIN:
FOO:
   ?*d1* = 1
   ?*d2* = 2
BAR:
   ?*d2* = 2
   ?*d3* = 3
CLIPS> (undefglobal BOGUS::d2)
[PRNTUTIL1] Unable to find defglobal BOGUS::d2.
CLIPS> (undefglobal FOO::d2)
CLIPS> (list-defglobals *)
MAIN:
FOO:
   d1
BAR:
   d2
   d3
For a total of 3 defglobals.
CLIPS> (undefglobal BAR::d2)
CLIPS> (list-defglobals *)
MAIN:
FOO:
   d1
BAR:
   d3
For a total of 2 defglobals.
CLIPS> (undefglobal *)
CLIPS> (list-defglobals *)
MAIN:
FOO:
   d1
BAR:
For a total of 1 defglobal.
CLIPS> (dribble-off)
